File: Programação_Funcional/Listas/Estrutura_Condicional/Ex1045.clj
(defn tipo-triangulo [a b c]
  (let [lados (sort > [a b c])
        a (first lados)
        b (second lados)
        c (last lados)]
    (if (or (>= a (+ b c)) (<= (+ b c) a))
      (println "NAO FORMA TRIANGULO")
      (do
        (println "TRIANGULO ACUTANGULO")
        (if (= (* a a) (+ (* b b) (* c c)))
          (println "TRIANGULO RETANGULO"))
        (if (> (* a a) (+ (* b b) (* c c)))
          (println "TRIANGULO OBTUSANGULO"))
        (if (= a b c)
          (println "TRIANGULO EQUILATERO")
          (if (or (= a b) (= b c) (= a c))
            (println "TRIANGULO ISOSCELES")))))))

(let [a (read)
      b (read)
      c (read)]
  (tipo-triangulo a b c))

File: Estrutura_Condicional/Python/Ex1045.py
a, b, c = map(float, input().split())

lados = [a, b, c]
lados.sort(reverse=True)

a, b, c = lados

if a >= b + c:
    print("NAO FORMA TRIANGULO")
else:
    if a**2 == b**2 + c**2:
        print("TRIANGULO RETANGULO")

    if a**2 > b**2 + c**2:
        print("TRIANGULO OBTUSANGULO")

    if a**2 < b**2 + c**2:
        print("TRIANGULO ACUTANGULO")

    if a == b == c:
        print("TRIANGULO EQUILATERO")

    if a == b != c or a == c != b or b == c != a:
        print("TRIANGULO ISOSCELES")